plugins/user_mapping/src/user_claim.cpp

```diff
@@ -0,0 +1,64 @@
+#include "interface.h"
+#include "user_claim.hpp"
+
+#include <cstdlib>
+#include <cstring>
+#include <exception>
+#include <string>
+
+#include <spdlog/spdlog.h>
+
+namespace
+{
+	std::string claim_to_match;
+} // anonymous namespace
+
+auto init(const nlohmann::json& _config) -> void {
+	const auto claim{_config.find("irods_user_claim")};
+	if (claim == std::end(_config)) {
+		// handle bad case
+		spdlog::error("{}: Unable to find 'irods_user_claim' in provided config.", __func__);
+	}
+
+	claim_to_match = *claim;
+}
+
+auto match(const nlohmann::json& _params) -> std::optional<std::string> {
+	if (auto claim{_params.find(claim_to_match)}; claim != std::end(_params)) {
+		return {*claim};
+	}
+
+	return std::nullopt;
+}
+
+void user_mapper_init(const char* _args) {
+	try {
+		spdlog::debug("{}: recieved _args [{}]", __func__, _args);
+		init(nlohmann::json::parse(_args));
+	} catch (const std::exception& e) {
+		spdlog::debug("{}: {}", __func__, e.what());
+	}
+}
+
+char* user_mapper_match(const char* _param) {
+	spdlog::debug("{}: Attempting match of _param [{}]", __func__, _param);
+	try {
+		auto res{match(nlohmann::json::parse(_param))};
+
+		if (res) {
+			spdlog::debug("{}: matched _param [{}] to user [{}]", __func__, _param, *res);
+			char* matched_username{static_cast<char*>(std::malloc(res->size()))};
+			std::strncpy(matched_username, res->c_str(), res->size() + 1);
+			return matched_username;
+		}
+	} catch (const std::exception& e) {
+		spdlog::debug("{}: {}", __func__, e.what());
+	}
+	return nullptr;
+}
+
+void user_mapper_close() {
```

Comment:
`match` and `init` use the `auto func() -> return_type` form.

Reply:
Done.